GANDLF/cli/huggingface_hub_handler.py

```diff
@@ -0,0 +1,52 @@
+from huggingface_hub import HfApi, snapshot_download
+from typing import List, Union
+
+
+def push_to_model_hub(
+    repo_id: str,
+    folder_path: str,
+    path_in_repo: Union[str, None] = None,
+    commit_message: Union[str, None] = None,
+    commit_description: Union[str, None] = None,
+    token: Union[str, None] = None,
+    repo_type: Union[str, None] = None,
+    revision: Union[str, None] = None,
+    allow_patterns: Union[List[str], str, None] = None,
+    ignore_patterns: Union[List[str], str, None] = None,
+    delete_patterns: Union[List[str], str, None] = None,
+):
+    api = HfApi(token=token)
+
+    api.create_repo(repo_id, exist_ok=True)
+
+    api.upload_folder(
+        repo_id=repo_id,
+        token=token,
+        folder_path=folder_path,
+        path_in_repo=path_in_repo,
+        commit_message=commit_message,
+        commit_description=commit_description,
+        repo_type=repo_type,
+        revision=revision,
+        allow_patterns=allow_patterns,
+        ignore_patterns=ignore_patterns,
+        delete_patterns=delete_patterns,
+    )
+
+
+def download_from_hub(
```

Comment:
From an external pair of eyes, I'm not sure to understand why defining `download_from_hub` which end-up being an alias for `snapshot_download`?

Reply:
@Wauplin, i did this to stay consistent with the implementations of other cli subcommands.

Reply:
Fine with it!

Reply:
@sarthakpati, the HF CLI offers more features than this implementation. However, incorporating a CLI subcommand for HF Hub might encourage more users to fully utilize it. Maybe we could explore integrating the HF CLI into our CLI subcommands if we require all its features.

Reply:
@pranayasinghcsmpl - Even if we are offering something under the `gandlf` family of commands, I think it should have the functionality of the HF CLI, since it seems like it serves the exact purposes that we need [[ref](https://github.com/mlcommons/GaNDLF/pull/880#discussion_r1629691344)]. What do you think?

Reply:
@sarthakpati, I think you're right. Also using the HF CLI might resolve issues mentioned in [ref](https://github.com/mlcommons/GaNDLF/pull/880#pullrequestreview-2101277500).

Reply:
Precisely! 

I am going to mark this PR as draft until this gets ready, @pranayasinghcsmpl. 

@NielsRogge & @Wauplin - would it be okay to tag you to take a look at this PR once @pranayasinghcsmpl is done from his end to provide additional feedback?

Reply:
Yes sounds good!

Reply:
Of course! :hugs:

Reply:
Thank you! 👍🏽